### PR TITLE
Fix as_rounded to produce decimal accurate results

### PR DESCRIPTION
**Pull request by carlism**
Using a bit of a hack to avoid needing a decimal module dependency

### CODE REVIEW

File: lib/crutches/format/number.ex [hunk 1]
```diff
@@ -167,7 +167,8 @@ defmodule Crutches.Format.Number do
     number |> make_significant(precision)
   end
   defp prepare_as_rounded(number, precision, false) do
-    number = Float.round(number, precision)
+    multiplier = :math.pow(10, precision)
+    number = number |> Kernel.*(multiplier) |> Float.round(0) |> Kernel./(multiplier)
```
Comment (duijf) [inline lib/crutches/format/number.ex]: I think the pipe operator makes this less readable. Can you change this line to:

```elixir
number = Float.round(multiplier * number, 0) / multiplier
```

**Review by duijf — CHANGES_REQUESTED** [review]
Hi @carlism! Thanks a lot for the PR :) 

I must admit I have no idea how this works exactly. Could you elaborate on your changes? I've also added a comment about style

**Comment by carlism** [comment]
Requested change made.  Thanks for the feedback.  And as for how it works, I noticed all the examples that round up to a whole number do it exactly as you'd expect.  So I simply shifted the decimal before the call to round so it was rounding to a whole number then shifted it back.  The alternative is to pull in a Decimal math dependency and do the rounding with decimal precision as opposed to floating point math.

**Review by duijf — APPROVED** [review]

**Comment by duijf** [comment]
Thanks for the explanation and for the contribution! :smile:

**Pull request merged**